pkgs/applications/audio/musescore/default.nix

```
@@ -140,6 +140,27 @@ in stdenv'.mkDerivation (finalAttrs: {
     ln -s $out/Applications/mscore.app/Contents/MacOS/mscore $out/bin/mscore
   '';
 
+  # muse-sounds-manager installs Muse Sounds sampler libMuseSamplerCoreLib.so.
+  # It requires that argv0 of the calling process ends with "/mscore" or "/MuseScore-4".
+  # We need to ensure this in two cases:
+  #
+  # 1) when the user invokes MuseScore as "mscore" on the command line or from
+  #    the .desktop file, and the normal argv0 is "mscore" (no "/");
+  # 2) when MuseScore invokes itself via File -> New, and the normal argv0 is
+  #    the target of /self/proc/exe, which in Nixpkgs was "{...}/.mscore-wrapped"
+  #
+  # In order to achieve (2) we install the final binary as $out/libexec/mscore, and
+  # in order to achieve (1) we use makeWrapper without --inherit-argv0.
+  #
+  # wrapQtAppsHook uses wrapQtApp -> wrapProgram -> makeBinaryWrapper --inherit-argv0
+  # so we disable it and explicitly use makeQtWrapper.
+  dontWrapQtApps = stdenv.isLinux;
+  postFixup = lib.optionalString stdenv.isLinux ''
```

Didn't we agree that this should be done for Darwin as well?

```suggestion
  dontWrapQtApps = true;
  postFixup = ''
```

We need to check somehow though if a similar action needs to be done in `$out/Applications`. I have access to the Darwin build box at `darwin-build-box.nix-community.org` but apparently musescore is broken on Darwin since the last build:

https://cache.nixos.org/log/28xjqwnby06ihpjqc0yq8alnqprai8p2-musescore-4.4.1.drv

https://hydra.nixos.org/build/272681576

---

We will also need someone to test how MuseScore behaves on macOS with respect to Muse Sounds. For example it is possible that libMuseSamplerCoreLib on macOS does not check the executable name, then Muse Sounds should already work there and this PR will not need further adjustments. I'd like to keep this PR Linux-only.

---

> We will also need someone to test how MuseScore behaves on macOS with respect to Muse Sounds. For example it is possible that libMuseSamplerCoreLib on macOS does not check the executable name, then Muse Sounds should already work there and this PR will not need further adjustments. I'd like to keep this PR Linux-only.

OK That's an option, but let's write a TODO comment for that.

---

In the meantime the 4.4.2 build of musescore on Darwin seems to go on well (it was probably an upstream regression in 4.4.1).

---

Added TODO